client/src/components/Search/SearchBox.js

```diff
@@ -292,60 +284,64 @@ class SearchBox extends Component {
     >
       <span className={spanClass}>{i18n._t('Admin.SEARCH_OPTIONS', 'Search options')}</span>
     </Button>);
-  }
+  };
 
   /**
    * Render the hide button.
    */
-  renderHideButton() {
-    const { id, onHide } = this.props;
-    return (<Button
-      onClick={onHide}
-      title={i18n._t('Admin.CLOSE_SEARCH', 'Close search')}
-      aria-label={i18n._t('Admin.CLOSE_SEARCH', 'Close search')}
-      className="btn--no-text btn--icon-lg search-box__cancel"
-      icon="cancel"
-      aria-controls={id}
-      aria-expanded="true"
-    />);
-  }
-
-  render() {
-    const { children, id, hideable, expanded, showFilters, dirty, clearable } = this.props;
-
-    const searchClasses = classNames('search-box', {
-      'search-box--hideable': hideable,
-      'search-box--not-hideable': !hideable,
-      'search-box--has-focus': this.state.hasFocus,
-      'search-box--has-not-focus': !this.state.hasFocus,
-      'search-box--has-filters': showFilters,
-      'search-box--has-not-filters': !showFilters,
-      'search-box--compact': this.state.width < 576, // 'sm' breakpoint
-      'search-box--expanded': expanded
-    });
-
-    const showEnter = (dirty || !clearable) && this.state.hasFocus;
-
-    return (
-      <div className={searchClasses} ref={node => { this.nodeRef = node; }}>
-        <ResizeAware onResize={this.onResize}>
-          <div className="search-box__group">
-            <Label for={id} id={`${id}_label`} hidden>
-              {i18n._t('Admin.SEARCH', 'Search')}
-            </Label>
-            { this.renderTags() }
-            { this.renderInput() }
-            { showEnter && this.renderEnterHint() }
-            { children }
-            <div className="icon font-icon-search" aria-hidden="true" />
-            { showFilters && this.renderFilterButton() }
-            { hideable && this.renderHideButton() }
-          </div>
-        </ResizeAware>
-      </div>
-    );
-  }
-}
+  const renderHideButton = () => (<Button
+    onClick={onHide}
+    title={i18n._t('Admin.CLOSE_SEARCH', 'Close search')}
+    aria-label={i18n._t('Admin.CLOSE_SEARCH', 'Close search')}
+    className="btn--no-text btn--icon-lg search-box__cancel"
+    icon="cancel"
+    aria-controls={id}
+    aria-expanded="true"
+  />);
+
+  useEffect(() => {
+    // Do not run on first render to match previous behaviour from class component
+    // Note: This hook has no dependencies so will behave like componentDidMount() used to
```

Comment:
There was no `componentDidMount` - there was only `componentDidUpdate`. Is that what you mean?

Reply:
Also, there's no mention in the class component implementation of not running in the first render - can you please explain why we need to explicitly manage that now?

Reply:
Yes `componentDidUpdate` is what I meant, have updated comment

In the class component this code did not run in the first render. We need the `isFirstRender.current` check to ensure that it does not run in the first render here.

Reply:
I still don't think this additional complexity is strictly needed but it's not worth pingpong so leave as is.